projects/admin/src/app/classes/typeahead/mef-typeahead.ts

```diff
@@ -81,20 +109,72 @@ export class MefTypeahead implements ITypeahead {
-          const label = data?.metadata?.authorized_access_point;
-          return `
-            <a href="${link}" target="_blank">${label} <i class="fa fa-external-link"></i></a>
-            <small class="badge badge-secondary ml-1">${source.toUpperCase()}</small>
-          `
+          const entity = {
+            uri: this._get_source_uri(data?.metadata?.identifiedBy, source) || value,
+            label: data?.metadata?.authorized_access_point
+          };
+          const badges = [{label: source.toUpperCase()}];
+          return this._buildEntityResolutionResponse(entity, badges);
         }),
-        catchError(e => {
-          console.error(`getValueAsHTML :: Unable to resolve ${value}`, e);
-          return of(`<span class="text-warning"><i class="fa fa-exclamation-triangle mr-1"></i>${value}</span>`);
-        })
+        catchError(e => this._handleEntityResolutionError(e, value))
     );
   }
 
+  /**
+   * Get HTML representation for a local entity value.
+   *
+   * @param options: typeahead options
+   * @param value: formControl value i.e. $ref value
+   * @returns Observable on HTML template representation of the value.
+   */
+  private _getLocalValueAsHTML(options: any, value: string): Observable<string> {
+    const pid = value.split('/').slice(-1)[0];
+    return this._recordService
+      .getRecord('local_entities', pid)
+      .pipe(
+         map((data: any) => {
+           const entity = {label: data.metadata.authorized_access_point};
+           const badges = [];
+           if (data.metadata?.source_catalog) {
+             badges.push({label: data.metadata.source_catalog});
+           }
+           return this._buildEntityResolutionResponse(entity, badges);
+         }),
+         catchError(e => this._handleEntityResolutionError(e, value))
+      );
+  }
+
+  /**
+   * Build the HTML representation for an entity.
+   * @param entity: the entity to represent.
+   * @param badges: badges list to append to the entity.
+   * @returns the HTML representation for the entity.
+   */
+  private _buildEntityResolutionResponse(entity: {label:string, uri?: string}, badges: Array<{label: string, type?: string}>): string {
+    let output = (entity?.uri)
+      ? `<a href="${entity.uri}" target="_blank">${entity.label} <i class="fa fa-external-link"></i></a>`
+      : `<span>${entity.label}</span>`;
+    badges.forEach(badge => {
+      const className = (badge?.type)
+        ? `badge-${badge.type}`
+        : 'badge-secondary';
+      output += `<small class="badge ${className} ml-1">${badge.label}</small>`;
+    });
+    return output;
+  }
+
+  /**
+   * Manage entity resolution error.
+   * @param error: the triggered error
+   * @param value: the entity value causing the resolution error.
+   */
+  private _handleEntityResolutionError(error: Error, value: string): Observable<string> {
+    console.error(`getValueAsHTML :: Unable to resolve entity :: ${value}`, error);
```

Comment:
Remove ?

Reply:
No. It sounds good (for me ?) to see an error in the console in this case.